Geometry2d/Point.hpp

```diff
@@ -40,6 +41,11 @@ class Point {
     Point(const QPoint& other) : Point(other.x(), other.y()) {}
 
     /**
+     * Implicit constructor for creating a Point from a float*
+     */
+    Point(double* other) : Point(double(other[0]), double(other[1])) {}
```

Comment:
What exactly is the use case of this? Could this be handled outside of the class?
If not, change to 
Point(const double* other) : Point(other[0],other[1]) {}